### PR TITLE
VACMS-18439 Facility Locator no content updates

### PR DESCRIPTION
**Pull request by randimays**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)

### :warning: TeamSites :warning:
Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
Update the content on the Facility Locator when no results are found. Per [the Figma design](https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=172-16230&node-type=canvas&t=jzwHmzbKBLDe4g7H-0), the text should be the same on desktop and on mobile (map view only). On mobile list view, a short error message should appear.

Note that this ticket also includes fixing the actual display of the error message. Previously the "no results" error message would show as soon as you select a facility type. It didn't require that you click the Search button. With these changes, it is now required that the search actually take place before showing a "no results" error message.

Also includes:
- Some style adjustments to more closely align with the latest design comp, but many things need adjustment so this is not intended to fix everything
- Adding a great deal of PropTypes to quiet down a bazillion lint warnings

## Related issue(s)
Ticket: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18439

## Testing done
Tested Juneau, AK for VA cemeteries for a "no results" scenario. Tested a few other scenarios searching for facilities and providers using various locations for regression testing.

## Screenshots

**Note**: The feedback button is not showing up in my screenshots because I was only running vets-website locally (and that button comes from content-build).

### Mobile

<details><summary>List view</summary>

#### Very small font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-21 at 4 10 00 PM     |   Screenshot 2024-10-21 at 4 09 50 PM    |

#### Small font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-21 at 4 11 22 PM     |   Screenshot 2024-10-21 at 4 11 13 PM   |

#### Medium font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-21 at 4 08 46 PM     |   Screenshot 2024-10-21 at 4 08 39 PM    |

#### Large font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-21 at 4 12 59 PM    |   Screenshot 2024-10-21 at 4 12 51 PM    |

#### Very large font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-21 at 4 14 42 PM    |   Screenshot 2024-10-21 at 4 14 31 PM   |

</details>

<details><summary>Map view</summary>

#### Very small font size
**Note**: The spacing between the map and the first line of text is collapsed on smaller text settings because of a [DST bug](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3421).

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-21 at 4 17 44 PM    |   Screenshot 2024-10-21 at 4 17 32 PM    |

#### Small font size
**Note**: The spacing between the map and the first line of text is collapsed on smaller text settings because of a [DST bug](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3421).

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-22 at 8 34 20 AM    |   Screenshot 2024-10-22 at 8 34 12 AM    |

#### Medium font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-22 at 8 34 41 AM    |   Screenshot 2024-10-22 at 8 34 34 AM    |

#### Large font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-22 at 8 35 09 AM    |   Screenshot 2024-10-22 at 8 35 00 AM    |

#### Very large font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-22 at 8 36 13 AM   |   Screenshot 2024-10-22 at 8 36 06 AM    |

</details>

#### Very small font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-22 at 8 40 42 AM    |   Screenshot 2024-10-22 at 8 40 35 AM   |

#### Small font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-22 at 8 40 59 AM   |   Screenshot 2024-10-22 at 8 40 51 AM   |

#### Medium font size

| Before | After |
| ------ | ----- |
|  Screenshot 2024-10-22 at 8 41 15 AM   |   Screenshot 2024-10-22 at 8 41 08 AM  |

#### Large font size

| Before |
| ------ |
|  Screenshot 2024-10-22 at 8 41 33 AM |

| After |
|  ----- |
|  Screenshot 2024-10-22 at 8 41 25 AM |

#### Very large font size

| Before |
| ------ |
|  Screenshot 2024-10-22 at 8 41 44 AM |

| After |
|  ----- |
|  Screenshot 2024-10-22 at 8 41 50 AM |

## Acceptance criteria
- [x] No results content has been updated for all scenarios to match Figma file
- [x] All text is wrapped in semantic tags so that the font resizes and scales appropriately
- [x] No results message does not display unless the user has activated the search button
- [x] There are no regressions in the tabindex nor focus management for no results message or the results listings for any of the Facility types
- [x] Requires design review
- [ ] Requires accessibility review

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed (by Randi)

### CODE REVIEW

File: src/applications/facility-locator/reducers/searchQuery.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import {
   SEARCH_STARTED,
   SEARCH_FAILED,
-  SEARCH_COMPLETE,
```
Comment (randimays) [inline src/applications/facility-locator/reducers/searchQuery.js:4]: There were no actions using this action type so I deleted it.

File: src/applications/facility-locator/utils/actionTypes.js [hunk 2]
```diff
@@ -10,7 +10,6 @@ export const CLEAR_SEARCH_TEXT = 'CLEAR_SEARCH_TEXT';
 
 export const SEARCH_STARTED = 'SEARCH_STARTED';
 export const SEARCH_FAILED = 'SEARCH_FAILED';
-export const SEARCH_COMPLETE = 'SEARCH_COMPLETE';
```
Comment (randimays) [inline src/applications/facility-locator/utils/actionTypes.js:13]: There were no actions using this action type so I deleted it.

File: src/applications/facility-locator/types/index.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
+import PropTypes from 'prop-types';
```
Comment (randimays) [inline src/applications/facility-locator/types/index.js:1]: We had so many lint warnings about missing prop types. So I pseudo-Typescript'd the types.

**Review by eselkin — APPROVED** [review]
LGTM. I like the reorganization. Tested on RI.

**Review by chriskim2311 — APPROVED** [review]
LGTM, tested in RI. Nice work on the prop re-org.

**Review by laflannery — APPROVED** [review]
Looks good!

**Review by CBonade — APPROVED** [review]

**Review by thejordanwood — APPROVED** [review]
LGTM!

**Comment by jilladams** [comment]
Fully approved and passing CI, so I'm merging for Randi since she was out today.

**Pull request merged**